dist/tools/riot_hdr/common.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU General Public
+ * License v2. See the file LICENSE for more details.
+ */
+
+#ifndef COMMON_H
+#define COMMON_H
+
+#include <unistd.h>
+#include <sys/types.h>
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+int to_file(const char *filename, void *buf, size_t len);
```

Comment:
I'd like to keep it in common since more functions will come. See #9969.

I'll document it but I'm not sure documentation for dist/tools is created.

Reply:
Generated to `docs.riot-os.org` or not, as a developer I am reading it.

From #9969 actually only the file management functions will be kept.

The `sha256` thing should be done by `riot_hdr_validate` or this file is doing more than just handling `riot_hdr`.
Or a `riot_hdr_validate_with_signature` if the header signature is not re-checked by the bootloader in `riot_hdr_validate`. But in that case, I would see the `OTA` mechanism being merged with the bootloader handling.